File: 0198-house-robber/0198-house-robber.java
class Solution {
    public int rob(int[] nums) {
        if (nums.length == 1) {
            return nums[0];
        }

        int prev2 = nums[0];
        int prev1 = Math.max(nums[0], nums[1]);
        int curr = prev1; // current max value

        for (int i = 2; i < nums.length; i++) {
            curr = Math.max(prev2 + nums[i], prev1);
            
            prev2 = prev1;
            prev1 = curr;
        }
        return curr;
    }
}

File: 0198-house-robber/0198-house-robber.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        if len(nums) < 2:
            return nums[0]
            
        for i in range(2, len(nums)):
            nums[i] += max(nums[i-2], nums[i-3] if i-3 > -1 else 0)
        
        return max(nums[-2], nums[-1])